silnlp/nmt/translate.py

```diff
@@ -1,41 +1,132 @@
         features_list = [encode_sp(self._src_spp, self._insert_lang_tag(s, trg_iso)) for s in sentences]
-        translations = self._runner.infer_list(features_list, checkpoint_path=str(self._checkpoint_path))
+        translations = self._runner.infer_list(features_list, checkpoint_path=str(self.checkpoint_path))
         return decode_sp_lines(t[0] for t in translations)
 
     def _insert_lang_tag(self, text: str, trg_iso: Optional[str]) -> str:
-        if self._multiple_trg_langs:
+        if self._multiple_trg_isos:
             if trg_iso is None:
                 trg_iso = self._default_trg_iso
             return f"<2{trg_iso}> {text}"
         return text
 
 
+@dataclass
+class TranslationTask:
+    name: str
+    checkpoint: str = "last"
+    memory_growth: bool = False
+    clearml_queue: str = None
+
+    def __post_init__(self):
+        if self.checkpoint is None:
+            self.checkpoint = "last"
+        self.translator = None
+
+    def init_translation_task(self, experiment_suffix: str):
+
+        self.clearml = SILClearML(
+            self.name, self.clearml_queue, project_suffix="_infer", experiment_suffix=experiment_suffix
+        )
+        self.name = self.clearml.get_remote_name()
+
+        SIL_NLP_ENV.copy_experiment_from_bucket(
+            self.name, extensions=(".vocab", ".model", ".yml", "dict.src.txt", "dict.trg.txt")
+        )
+
+        self.config: Config = self.clearml.load_config()
+        self.config.set_seed()
+
+        checkpoint_path, step = get_checkpoint_path(self.config.model_dir, self.checkpoint)
+        self.translator = NMTTranslator(
+            config=self.config, checkpoint_path=checkpoint_path, memory_growth=self.memory_growth
+        )
+        self._step_str = "avg" if step == -1 else str(step)
+
+    def translate_book(
+        self,
+        book: str,
+        src_project: Optional[str] = None,
+        output_usfm: Optional[str] = None,
+        trg_iso: Optional[str] = None,
+    ):
+        self.init_translation_task(experiment_suffix=f"_{self.checkpoint}_{book}")
+        if src_project is None:
+            if len(self.config.src_projects) != 1:
+                raise RuntimeError("A source project must be specified.")
+            src_project = next(iter(self.config.src_projects))
+
+        default_output_dir = self.config.exp_dir / "infer" / self._step_str
+        output_path: Optional[Path] = None if output_usfm is None else Path(output_usfm)
+        if output_path is None:
+            book_num = book_id_to_number(book)
+            output_path = default_output_dir / f"{book_file_name_digits(book_num)}{book}.SFM"
+        elif output_path.name == output_path:
+            output_path = default_output_dir / output_path
+
+        output_path.parent.mkdir(exist_ok=True, parents=True)
+        self.translator.translate_book(src_project, book, output_path, trg_iso=trg_iso)
+        SIL_NLP_ENV.copy_experiment_to_bucket(self.name)
+
+    def translate_parts(
```

Comment:
Can you rename this to something more explicit? Maybe `translate_text_files`.

Reply:
Done.